sonar-ruby-plugin/src/main/java/org/sonarsource/ruby/converter/RubyConverter.java

```diff
@@ -137,12 +139,28 @@ private Tree parseContent(String content) {
       .collect(Collectors.toList());
     TreeMetaDataProvider metaDataProvider = new TreeMetaDataProvider(comments, tokens);
 
+    RubyVisitor rubyVisitor = getRubyProcessor(metaDataProvider);
+    Object[] astProcessorParams = {rubyParseResult.get(0)};
+    NodeAdapter node = (NodeAdapter) invokeMethod(rubyVisitor, "process", astProcessorParams);
+
     if (tokens.isEmpty() && comments.isEmpty()) {
       throw new ParseException("No AST node found");
     }
 
     TreeMetaData topTreeMetaData = metaDataProvider.metaData(getFullRange(tokens, comments));
-    return new TopLevelTreeImpl(topTreeMetaData, Collections.emptyList(), comments);
+    if (node == null || node.getTree() == null) {
+      // only comments
+      return new TopLevelTreeImpl(topTreeMetaData, emptyList(), comments);
+    } else {
+      // singleton expression: we wrap it around a top level tree
+      return new TopLevelTreeImpl(topTreeMetaData, singletonList(node.getTree()), comments);
+    }
+  }
+
+  private RubyVisitor getRubyProcessor(TreeMetaDataProvider metaDataProvider) {
```

Comment:
Should we rename this method to `getRubyVisitor` ?

Reply:
Yep, to be consistent I renamed all remaining "rubyProcessor" to "rubyVisitor"